command/agent/agent.go

```diff
@@ -135,11 +130,11 @@ func (a *Agent) serverConfig() (*nomad.Config, error) {
 	}
 
 	// Set up the bind addresses
-	rpcAddr, err := a.getRPCAddr(true)
+	rpcAddr, err := net.ResolveTCPAddr("tcp", a.config.Addresses.RPC)
```

Comment:
Mentioned offline that it isn't always the case that the client can resolve its advertise address